### PR TITLE
Fix userId order in LdapUserService

### PR DESCRIPTION
**Pull request by Shoothzj**
## What's the purpose of this PR
Fix userId order in LdapUserService

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/apolloconfig/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [x] Update the [`CHANGES` log](https://github.com/apolloconfig/apollo/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by nobodyiam** [comment]
Would you please help to describe the actual issue?

**Comment by nobodyiam** [comment]
review again

**Comment by nobodyiam** [comment]
Hello, I am a [serverless review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/). Here are my reviews of code commits in this PR.

------

### Commit 1
The patch fixes a bug in the LdapUserService class that previously did not sort the user IDs correctly. The patch modifies the searchUsers() method to sort the user IDs before returning them. Potential issues are that the patch may introduce new bugs, and the change may have unintended consequences on other parts of the codebase.

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/ldap/LdapUserService.java [hunk 1]
```diff
@@ -279,7 +279,7 @@ public List<UserInfo> searchUsers(String keyword, int offset, int limit, boolean
         if (o1.getUserId().equals(o2.getUserId())) {
           return 0;
         }
-        return -1;
+        return o1.getUserId().compareTo(o2.getUserId());
```
Comment (Anilople) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/ldap/LdapUserService.java:282]: simplify

```java
        if (o1.getUserId().equals(o2.getUserId())) {
          return 0;
        }
        return o1.getUserId().compareTo(o2.getUserId());
```
to
```java
(o1, o2) -> o1.getUserId().compareTo(o2.getUserId())
```

Is it better?

**Comment by nobodyiam** [comment]
review again

**Comment by nobodyiam** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/ldap/LdapUserService.java (hunk 1 above)
Comment (Shoothzj) [reply apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/ldap/LdapUserService.java:282]: @Anilople Fixed

**Pull request closed without merging**